test/groups/test_general_linear_group.jl

```diff
@@ -0,0 +1,70 @@
+using LieGroups, Random, Test
+
+using ManifoldsBase: ℂ
+
+s = joinpath(@__DIR__, "..", "LieGroupsTestSuite.jl")
+!(s in LOAD_PATH) && (push!(LOAD_PATH, s))
+using LieGroupsTestSuite
+
+@testset "GL(2)" begin
+    G = GeneralLinearGroup(2)
+    g1, g2, g3 = [2.0 0.0; 0.0 1.0], [1.0 0.5; 0.5 1.0], [1.0 2.0; 3.0 4.0]
+    X1, X2, X3 = [-1.0 0.0; 0.0 0.0], [0.0 0.5; 0.5 0.0], [1.0 2.0; 3.0 4.0]
+    properties = Dict(
+        :Name => "The general linear group",
+        :Points => [g1, g2, g3],
+        :Vectors => [X1, X2, X3],
+        :Rng => Random.MersenneTwister(),
+        :Functions => [
+            compose,
+            conjugate,
+            diff_conjugate,
+            diff_inv,
+            diff_left_compose,
+            diff_right_compose,
+            exp,
+            # hat, # requires a fix in Manifolds.jl to have an ONB on invertible matrices
+            inv,
+            inv_left_compose,
+            inv_right_compose,
+            is_identity,
+            lie_bracket,
+            log,
+            #rand, # requires a fix in Manifolds.jl to have rand on invertible matrices
```

Comment:
If we do something on invertible matrices we can as well check to pass on bases to Euclidean and maybe check whether we find a way to implement rand.

Reply:
Do you prefer to finish it in this PR or a new one?

Reply:
This is probably more something for Manifolds.jl anyways. so a PR there.